test/ZNSStringResolver.test.ts

```diff
@@ -0,0 +1,465 @@
+      "(It cannot rewrite AC address after an incorrect address has been submitted to it)", async () => {
+
+      await expect(
+        stringResolver.connect(admin).setAccessController(admin.address)
+      ).to.emit(
+        stringResolver, "AccessControllerSet"
+      ).withArgs(admin.address);
+
+      expect(
+        await stringResolver.getAccessController()
+      ).to.equal(admin.address);
+    });
+  });
+
+  describe("New campaign for each test", () => {
+
+    let deployer : SignerWithAddress;
+    let zeroVault : SignerWithAddress;
+    let operator : SignerWithAddress;
+    let user : SignerWithAddress;
+    let deployAdmin : SignerWithAddress;
+    let admin : SignerWithAddress;
+    let mockRegistrar : SignerWithAddress;
+
+    const env = "dev";
+
+    let stringResolver :  ZNSStringResolver;
+    let registry : ZNSRegistry;
+    let campaign : DeployCampaign<
+    HardhatRuntimeEnvironment,
+    SignerWithAddress,
+    DefenderRelayProvider,
+    IZNSContracts
+    >;
+    let accessController : ZNSAccessController;
+    let domainToken : ZNSDomainToken;
+    let operatorBalance : bigint;
+    let userBalance : bigint;
+    let deployerBalance : bigint;
+
+    let mongoAdapter : MongoDBAdapter;
+
+    beforeEach(async () => {
+
+      [
+        deployer,
+        zeroVault,
+        user,
+        deployAdmin,
+        admin,
+        operator,
+        mockRegistrar,
+      ] = await hre.ethers.getSigners();
+
+      const campaignConfig = await getConfig({
+        deployer: deployer as unknown as SignerWithAddress,
+        governors: [deployAdmin.address],
+        admins: [admin.address],
+        zeroVaultAddress: zeroVault.address,
+        env,
+      });
+
+      campaign = await runZnsCampaign({
+        config: campaignConfig,
+      });
+
+      let meowToken : MeowTokenMock;
+      let treasury : ZNSTreasury;
+
+      ({
+        stringResolver,
+        registry,
+        meowToken,
+        treasury,
+        accessController,
+        domainToken,
+        dbAdapter: mongoAdapter,
+      } = campaign);
+
+      operatorBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(operator.address, operatorBalance);
+      await meowToken.connect(operator).approve(await treasury.getAddress(), ethers.MaxUint256);
+
+      userBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(user.address, userBalance);
+      await meowToken.connect(user).approve(await treasury.getAddress(), ethers.MaxUint256);
+
+      deployerBalance = ethers.parseEther("1000000000000000000");
+      await meowToken.mint(deployer.address, deployerBalance);
+      await meowToken.connect(deployer).approve(await treasury.getAddress(), ethers.MaxUint256);
+    });
+
+    afterEach(async () => {
+      await mongoAdapter.dropDB();
+    });
+
+    it("Should not allow non-owner address to setString (similar domain and string)", async () => {
```

Comment:
what does "(similar domain and string)" mean?

Reply:
Just different test params to cover more scenarios